### DISCUSSION TITLE
Populate multi-select picker with data output from automation

### DISCUSSION THREAD

**Discussion by pcgeek86** [Budibase/budibase#11092] [Help] [answered]
### Scenario

- I have a form with a multi-select picker, and a button.
- Clicking the button triggers an automation.
- The automation executes a Bash Script, which retrieves some data.

### Objective

I want to use the data returned from the automation to populate the list items in the multi-select picker.

How can I accomplish this?

**Comment by R2bEEaton** [comment, marked as answer]
Users with at least a Business license can receive results back from Automations on the front-end. However, if the Bash Script itself is not critical, here are some possible solutions to explore:

### Store the result in a table
A possible solution would be to add another step to your Automation that stores the result of your Bash Script in a new table. You can then use this table as the data source for your multi-select picker using [the process explained here](https://docs.budibase.com/docs/option-picker#displaying-options-in-the-options-picker). Make sure to reload the data source after pushing the button using the "Refresh data provider" button action for the new data to show up in the picker.

### Direct data source / custom query
Alternatively, if possible, cut out the middle man and convert your Bash Script into a direct data source in Budibase. Budibase supports [a lot of data sources](https://docs.budibase.com/docs/data-sources) and, combined with [custom query transformers](https://docs.budibase.com/docs/transformers), you might be able to achieve the same effect as your Bash Script. This would also be beneficial because the data would be live. Every time you load the page with the multi-select picker, it will automatically pull the latest data from your custom query transformer. In my opinion, and of course depending on your use case, this would be the best option.

### Hope this helps!
Let me know if you have any further questions! If this helped, make sure to mark it answered so it can help future users! 😃

*Edited to reflect the correction from @PClmnt's comment.*

**Reply by PClmnt** [reply]
@R2bEEaton Great answer, just a heads up that for Business customers and above, there is now the ability to retrieve automation results to the front end: https://docs.budibase.com/docs/synchronous-automations
